pkg/provider/beyondtrust/provider.go

```diff
@@ -0,0 +1,352 @@
+		}
+		if ref.SecretRef.Name == "" {
+			return errMissingSecretName
+		}
+		if ref.SecretRef.Key == "" {
+			return errMissingSecretKey
+		}
+	} else if ref.Value == "" {
+		return errSecretRefAndValueMissing
+	}
+
+	return nil
+}
+
+func (p *Provider) GetAllSecrets(_ context.Context, _ esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	return nil, fmt.Errorf("GetAllSecrets not implemented")
+}
+
+// GetSecret reads the secret from the Password Safe server and returns it. The controller uses the value here to
+// create the Kubernetes secret.
+func (p *Provider) GetSecret(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	managedAccountType := !strings.EqualFold(p.retrievaltype, "SECRET")
+
+	retrievalPaths := utils.ValidatePaths([]string{ref.Key}, managedAccountType, p.separator, &p.log)
+
+	if len(retrievalPaths) != 1 {
+		return nil, fmt.Errorf(errInvalidRetrievalPath)
+	}
+
+	retrievalPath := retrievalPaths[0]
+
+	_, err := p.authenticate.GetPasswordSafeAuthentication()
+	if err != nil {
+		return nil, fmt.Errorf("error: %w", err)
+	}
+
+	secret := keyValue{
+		Key:   "secret",
+		Value: "",
+	}
+
+	if managedAccountType {
+		var returnSecret string
+		ESOLogger.Info("retrieve managed account value", "retrievalPath:", retrievalPath)
+		manageAccountObj, _ := managed_account.NewManagedAccountObj(p.authenticate, &p.log)
+		returnSecret, err := manageAccountObj.GetSecret(retrievalPath, p.separator)
+		if err != nil {
+			err = p.authenticate.SignOut()
+			if err != nil {
+				return nil, fmt.Errorf("error: %w", err)
+			}
+			return nil, fmt.Errorf("error: %w", err)
+		}
+		secret.Value = returnSecret
+	} else {
+		var returnSecret string
+		ESOLogger.Info("retrieve secrets safe value", "retrievalPath:", retrievalPath)
+		secretObj, _ := secrets.NewSecretObj(p.authenticate, &p.log, maxFileSecretSizeBytes)
+		returnSecret, err := secretObj.GetSecret(retrievalPath, p.separator)
+		if err != nil {
+			err = p.authenticate.SignOut()
+			if err != nil {
+				return nil, fmt.Errorf("error: %w", err)
+			}
+			return nil, fmt.Errorf("error: %w", err)
+		}
+		secret.Value = returnSecret
+	}
+
+	return []byte(secret.Value), nil
+}
+
+// ValidateStore validates the store configuration to prevent unexpected errors.
+func (p *Provider) ValidateStore(store esv1beta1.GenericStore) (admission.Warnings, error) {
+	if store == nil {
+		return nil, fmt.Errorf(errNilStore)
+	}
+
+	spec := store.GetSpec()
+
+	if spec == nil {
+		return nil, fmt.Errorf(errMissingStoreSpec)
+	}
+
+	if spec.Provider == nil {
+		return nil, fmt.Errorf(errMissingProvider)
+	}
+
+	provider := spec.Provider.Beyondtrust
+	if provider == nil {
+		return nil, fmt.Errorf(errInvalidProvider, store.GetObjectMeta().String())
+	}
+
+	apiURL, err := url.Parse(provider.APIURL)
+	if err != nil {
+		return nil, fmt.Errorf(errInvalidHostURL)
+	}
+
+	if apiURL.Host == "" {
+		return nil, fmt.Errorf(errInvalidHostURL)
+	}
+
+	return nil, nil
+}
```

Comment:
I take if `clientID` or `clientSecret` are not set, you also want to validate them here, no? As in shouldn't these manifests be rejected by the validating webhook?

Reply:
I'm Validating them later on in the flow:

https://github.com/external-secrets/external-secrets/pull/3683/files#diff-28b45865eb17dee9472bdbdb0cccc3ca3f6003fb6c3c996c9aa27f1207e38e67R142

Reply:
I know - but these are different validation types. `ValidateStore` is used on the validating webhook - it prevents manifests without the configuration from even being created. IMO - this check needs to be added here.

Reply:
Validation was included, please check

Reply:
Hello @gusfcarvalho, I've solved comments, Could you please take a look at them ? Thank you so much.